src/coreComponents/physicsSolvers/fluidFlow/wells/CompositionalMultiphaseWell.cpp

```diff
@@ -333,37 +333,36 @@ void CompositionalMultiphaseWell::validateWellConstraints( MeshLevel const & mes
                                                                WellElementSubRegion const & subRegion )
   {
     WellControls const & wellControls = getWellControls( subRegion );
-    WellControls::Type const wellType = wellControls.getType();
     WellControls::Control const currentControl = wellControls.getControl();
     real64 const & targetTotalRate = wellControls.getTargetTotalRate( m_currentTime + m_currentDt );
     real64 const & targetPhaseRate = wellControls.getTargetPhaseRate( m_currentTime + m_currentDt );
     integer const useSurfaceConditions = wellControls.useSurfaceConditions();
     real64 const & surfaceTemp = wellControls.getSurfaceTemperature();
 
-    GEOSX_THROW_IF( wellType == WellControls::Type::INJECTOR && currentControl == WellControls::Control::PHASEVOLRATE,
+    GEOSX_THROW_IF( wellControls.isInjector() && currentControl == WellControls::Control::PHASEVOLRATE,
```

Comment:
Depending on how you feel it, you may want to reproduce the pattern for `currentControl == WellControls::Control::PHASEVOLRATE` too 😉